daprdocs/content/en/developing-applications/crewai/crewai-workflows.md

```diff
@@ -0,0 +1,204 @@
+---
+type: docs
+title: "CrewAI Workflows"
+linkTitle: "CrewAI Workflows"
+weight: 25
+description: "How to run CrewAI agents with durable, fault-tolerant execution using Dapr Workflows"
+---
+
+## Overview
+
+Dapr Workflows make it possible to run CrewAI agents **reliably**, **durably**, and **with built-in resiliency**.  
+By orchestrating CrewAI tasks with the Dapr Workflow engine, developers can:
+
+- Ensure long-running CrewAI work survives crashes and restarts  
+- Get automatic checkpoints, retries, and state recovery  
+- Run each CrewAI task as a durable activity  
+- Observe execution through tracing, metrics, and structured logs  
+
+This guide walks through orchestrating multiple CrewAI tasks using Dapr Workflows, ensuring each step is run *exactly once* even if the process restarts.
+
+## Getting Started
+
+Initialize Dapr locally to set up a self-hosted environment for development. This process installs the Dapr sidecar binaries, provisions the workflow engine, and prepares a default components directory. For full details, see the official [guide on initializing Dapr locally]({{% ref install-dapr-selfhost.md %}}).
+
+Initialize Dapr:
+
+```bash
+dapr init
+```
+
+Verify that daprio/dapr, openzipkin/zipkin, and redis are running:
+
+```bash
+docker ps
+```
+
+### Install Python
+
+{{% alert title="Note" color="info" %}}
+Make sure you have Python already installed. `Python >=3.10`. For installation instructions, visit the official [Python installation guide](https://www.python.org/downloads/).
+{{% /alert %}}
+
+### Install Dependencies
+
+```bash
+pip install dapr dapr-ext-workflow crewai
+```
+
+### Create a Python Virtual Environment (recommended)
+
+```bash
+python -m venv .venv
+source .venv/bin/activate     # Windows: .venv\Scripts\activate
+```
+
+### Create a Workflow to Run CrewAI Tasks
+
+Create a file named crewai_workflow.py and paste the following:
```

Comment:
Provide an overview explanation of what this application does, to prepare reading the code